rust/datafusion/src/sql/planner.rs

```diff
@@ -628,6 +701,53 @@ impl<'a, S: SchemaProvider> SqlToRel<'a, S> {
     }
 }
 
+fn create_join_schema(left: &SchemaRef, right: &SchemaRef) -> Result<Schema> {
+    let mut fields = vec![];
+    for field in left.fields() {
+        fields.push(field.clone());
+    }
+    for field in right.fields() {
+        fields.push(field.clone());
+    }
+    Ok(Schema::new(fields))
+}
+
+/// Parse equijoin ON condition which could be a single Eq or multiple conjunctive Eqs
+///
+/// Examples
+///
+/// foo = bar
+/// foo = bar AND bar = baz AND ...
+///
+fn extract_join_keys(expr: &Expr, accum: &mut Vec<(String, String)>) -> Result<()> {
+    match expr {
```

Comment:
👍 

When I was reading this I have the feeling that the general pattern of "take an expression tree and break it into a list of conuncts (aka things that are AND'd together) will be a common one as we split filters up and push them around. 

No action required here, just simply an observation